src/extension/src/TelemetryWriter.py

```diff
@@ -65,17 +66,23 @@ def __ensure_message_restriction_compliance(self, full_message):
             self.logger.log_telemetry_module_error("Error occurred while formatting message for a telemetry event. [Error={0}]".format(repr(e)))
             raise
 
-    @staticmethod
-    def __get_agent_supports_telemetry_from_env_var():
+    def __get_agent_supports_telemetry_from_env_var(self):
         """ Returns True if the env var AZURE_GUEST_AGENT_EXTENSION_SUPPORTED_FEATURES has a key of
             ExtensionTelemetryPipeline in the list. Value of the env var looks like this:
             '[{  "Key": "ExtensionTelemetryPipeline", "Value": "1.0"}]' """
         features_keyvalue_list_str = os.getenv(Constants.AZURE_GUEST_AGENT_EXTENSION_SUPPORTED_FEATURES_ENV_VAR)
         if features_keyvalue_list_str is None:
+            self.agent_env_var_code = 1
             return False
 
         features_keyvalue_list = json.loads(features_keyvalue_list_str)
-        return any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        telemetry_supported_key_exists = any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        if telemetry_supported_key_exists is False:
+            self.logger.log_error('Guest agent does not support telemetry. [Error=Key not found: {0}]'.format(Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY))
+            self.logger.log_warning('Env var value: \n{0}'.format(json.dumps(features_keyvalue_list, indent=4, sort_keys=True)))
+            self.agent_env_var_code = 2
```

Comment:
Same comment about Enum -- look at PatchOperationErrorCodes as an example for what was meant.